### PR TITLE
fix(pod-controller): sync pod label into endpoint

### PR DESCRIPTION
**Pull request by wu-cl**
endpoint selector in SecurityPolicy will select corresponding endpoints according to labels in
endpoint.

### CODE REVIEW

File: pkg/controller/k8s/pod_controller.go [hunk 1]
```diff
@@ -70,28 +70,50 @@ func (r *PodReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
+		// submit update
+		if err := r.Update(ctx, &endpoint); err != nil {
+			klog.Errorf("update endpoint %s err: %s", endpointName, err)
+			return ctrl.Result{}, err
+		}
+	}
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("Get endpoint error, err: %s", err)
 		return ctrl.Result{}, err
 	}
```
Comment (KeyongSun) [inline pkg/controller/k8s/pod_controller.go:118]: if err == nil
else if errors.IsNotFound(err)
else {}
return

Comment (wu-cl) [reply pkg/controller/k8s/pod_controller.go:118]: if else is forbidden in golint, which suggests to use switch. 
So, this part has to split into three if.

Comment (zwtop) [reply pkg/controller/k8s/pod_controller.go:118]: you can return at the end of if.
```
if err == nil {
    // handle
    return nil
}
if errors.IsNotFound(err) {
    // handle
    return nil
}
return err
```

**Review by zwtop — COMMENTED** [review]
need update pod labels test cases.

File: pkg/controller/k8s/pod_controller.go (hunk 1 above)
Comment (KeyongSun) [reply pkg/controller/k8s/pod_controller.go:118]: then we can use switch case, return is the same in all cases

**Review by KeyongSun — APPROVED** [review]

**Comment by wu-cl** [comment]
if else has been replaced with:
```
switch errors.ReasonForError(err) {
	case metav1.StatusReasonNotFound:
        case metav1.StatusReasonUnknown:
        defatut:
}
```
Test cases will be updated in #156

**Pull request merged**